### PR TITLE
Feature control surface derivatives

### PR DESCRIPTION
**Pull request by rachealerhard**
Adds control surface deflection derivatives to aero derivatives

### CODE REVIEW

File: trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py [hunk 1]
```diff
@@ -173,39 +170,25 @@ def compute_aero_derivatives(segment):
...
+            for cs in list(wing.control_surfaces):
+                delta             = segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces[cs].deflection                
+                perturbed_segment = deepcopy(segment)    
+                delta_plus        = delta + 0.1
+                perturbed_segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces[cs].deflection = delta_plus
+                
+                iterate = perturbed_segment.process.iterate
+                iterate.conditions(perturbed_segment)                 
+                
+                dDelta          = perturbed_segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces[cs].deflection - segment.analyses.aerodynamics.geometry.wings[wing].control_surfaces[cs].deflection
```
Comment (planes) [inline trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py:186]: This is much better!

However, can you also calculate the other aero derivatives Cl, Cn, Cm, etc..? That would so useful!

Comment (rachealerhard) [reply trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py:186]: Yes - added the others as well

File: trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py [hunk 2]
```diff
@@ -168,6 +168,81 @@ def compute_aero_derivatives(segment):
...
+                segment.state.conditions.aero_derivatives[dCn_dDelta_tag]  = dCn_dDelta
+                segment.state.conditions.aero_derivatives[dCl_dDelta_tag]  = dCl_dDelta
+                segment.state.conditions.aero_derivatives[dCT_dDelta_tag]  = dCT_dDelta
+                segment.state.conditions.aero_derivatives[dCP_dDelta_tag]  = dCP_dDelta
+
+    # ----------------------------------------------------------------------------    
+    # Velocity magnitude perturbation
+    vinf                = segment.state.conditions.frames.inertial.velocity_vector
+    vmag                = np.linalg.norm(vinf,axis=1)
+    psi                 = np.arctan2(vinf[:,2],vinf[:,0])
```
Comment (planes) [inline trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py:226]: I think this is gamma, the flight path angle. PSI is the heading angle

File: trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py [hunk 3]
```diff
@@ -168,6 +168,81 @@ def compute_aero_derivatives(segment):
...
+                # propeller derivatives 
+                dCT, dCP = propeller_derivatives(segment, perturbed_segment, n_cpts) 
+                    
+                dCL_dDelta      = dCL/dDelta 
+                dCD_dDelta      = dCD/dDelta
+                dCn_dDelta      = dCn/dDelta            
+                dCl_dDelta      = dCl/dDelta           
+                dCT_dDelta      = dCT/dDelta        
+                dCP_dDelta      = dCP/dDelta
+                dCL_dDelta_tag  = 'dCL_dDelta_'+cs
```
Comment (planes) [inline trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py:209]: Add dCm_dDelta

**Review by planes — APPROVED** [review]
Two comments to update. Then it's good to go

File: trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py (hunk 2 above)
Comment (rachealerhard) [reply trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py]: Done, thanks!

File: trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py (hunk 3 above)
Comment (rachealerhard) [reply trunk/SUAVE/Methods/Flight_Dynamics/Static_Stability/compute_aero_derivatives.py:212]: Done, thanks!

**Pull request merged**